conda/gateways/connection/adapters/jlap/repodata_proxy.py

```diff
@@ -114,65 +83,58 @@ def apply_patches(data, patches, have, want):
     return data
 
 
-@contextlib.contextmanager
-def timeme(message=""):
-    begin = time.time()
-    yield
-    end = time.time()
-    log.debug(f"{message}{end-begin:0.02f}s")
+def fetch_repodata_json(cache_path: Path, repodata_url: str, ses=None) -> Tuple[Path, str]:
+    """
+    Fetch new repodata.json; cache to a gzip'd file
+    """
+    if ses is None:
+        ses = requests
 
+    with tempfile.NamedTemporaryFile(dir=consts.CACHE_DIR, delete=False) as outfile:
+        with gzip.open(outfile, "w") as compressed:
+            response = ses.get(repodata_url, stream=True)
+            response.raise_for_status()
+            hash_val = hash_func()
 
-def fetch_repodata_json(cache_path, upstream, session=requests):
-    """
-    Fetch new repodata.json; cache to a gzip'd file.
+            for chunk in response.iter_content(consts.JLAP_CHUNK_SIZE):
```

Comment:
IIRC response.iter_content() with no arguments is requests' default 10kb chunk size